include/boost/math/tools/roots.hpp

```diff
@@ -275,14 +275,7 @@ T newton_raphson_iterate(F f, T guess, T min, T max, int digits, std::uintmax_t&
       if (fabs(delta * 2) > fabs(delta2))
       {
          // Last two steps haven't converged.
-         T shift = (delta > 0) ? (result - min) / 2 : (result - max) / 2;
-         if ((result != 0) && (fabs(shift) > fabs(result)))
-         {
-            delta = sign(delta) * fabs(result) * 1.1f; // Protect against huge jumps!
-            //delta = sign(delta) * result; // Protect against huge jumps! Failed for negative result. https://github.com/boostorg/math/issues/216
-         }
-         else
-            delta = shift;
+         delta = (delta > 0) ? (result - min) / 2 : (result - max) / 2;
```

Comment:
I looked into this. Here is the history...

#145 @jzmaddock opened an issue where skew-normal quantiles return non-finite values. This issue occurs because the min/max bounds for the 1D Newton search problem are -Inf/+Inf. Attempting to bisect the search point at either ends gets one stuck at either -Inf/+Inf.
91c193d2d2c1913f42692795beceb5252c18c09d @jzmaddock Fixed #145 by adding an early version of huge step protection. A test was also added to `test_roots` to prevent possible regressions.
3ab69d00eefdf8afb1db13a4aecea3a9e1f8f97f The test was removed as part of adding the complex Newton solver, possibly due to a merge conflict issue.
#422 @evanmiller added the kolmogorov-smirnov distribution. The min/max bounds for the 1D Newton search problem are 0/max_value<RealType>(). This new functionality requires huge step protection because evaluating the functor at large values causes the program to hang.

Reply:
Huge step protection seems to cover up issues with Newton search bounds for quantile problems.
 
This PR attempts to resolve this with the following steps:
1. Change the `skew_normal` quantile search to go from -max_value/+max_value instead of -Inf/+Inf.
2. Change the upper search bound of the kolmogorov-smirov distribution to `RealType(1)`. This fixes the hang issue, and causes no test issues, **I don't know if it is right though.**
3. Add the test that the skew-normal quantity is finite back in

Reply:
> Removing the large step protection is probably a bad idea, but I would have to go searching for a test case to tell you why ;)

There's a test case that requires it in the current code-base, although it isn't causing a test failure. Discussed further here: https://github.com/boostorg/math/pull/1000#issuecomment-1659036000.